### PR TITLE
Update count in-memory store only when Mount and Unmount operations have completed successfully

### PR DESCRIPTION
**Pull request by shishir-a412ed**
Signed-off-by: Shishir Mahajan <shishir.mahajan@redhat.com>

### CODE REVIEW

**Comment by shishir-a412ed** [comment]
ping @rhatdan

File: driver.go [hunk 1]
```diff
@@ -280,7 +281,7 @@ func (l *lvmDriver) Mount(req volume.MountRequest) volume.Response {
 		return false, ""
 	}()
 
-	if l.count[req.Name] == 1 {
```
Comment (rhatdan) [inline driver.go]: Why not 
if l.count[req.Name] == 0 {
}
l.count[req.Name]++

File: driver.go [hunk 2]
```diff
@@ -325,8 +327,9 @@ func (l *lvmDriver) Mount(req volume.MountRequest) volume.Response {
 func (l *lvmDriver) Unmount(req volume.UnmountRequest) volume.Response {
 	l.mu.Lock()
 	defer l.mu.Unlock()
-	l.count[req.Name]--
-	if l.count[req.Name] == 0 {
+	count := l.count[req.Name]
+	count--
+	if count == 0 {
```
Comment (rhatdan) [inline driver.go]: Why not 
if l.count[req.Name] == 1 {
}
l.count[req.Name] --

**Comment by rhatdan** [comment]
Does this relate to a bugzilla?

**Comment by shishir-a412ed** [comment]
@rhatdan No, but if there was an error in Mount and Unmount operations, the in-memory store was getting updated before the operations {Mount, Unmount} and before it could save the new state to the disk, it would error out. The in-memory store and on disk state would then get out of sync. This PR addresses that.

Shishir

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
Fix the commit message to match what you just said, and we can merge.

Do multi line

**Comment by shishir-a412ed** [comment]
@rhatdan Fixed.
Merging this PR.

Shishir

**Comment by rhatdan** [comment]
Fix first line so it does not go over multiple lines.
Something short  like

Update count in-memory store on successful mount/unmount

**Comment by shishir-a412ed** [comment]
@rhatdan Fixed.
PTAL.

Shishir

**Pull request merged**